### DISCUSSION TITLE
Ambient Weather ws-2902 sensor array setup help

### DISCUSSION THREAD

**Discussion by Sandman2001** [merbanan/rtl_433#3002] [Q&A] [answered]
I'm looking to implement a receiver station for the [Ambient Weather sensor array](https://ambientweather.com/amws2902array.html). does anyone know how this sensor transmits its data? Is it one conglomerated packet or does each sensor send its data individually?  This is my first time utilizing the rtl_433 library so I'm not familiar with the complexity of configuring sensors not directly listed as compatible in the rtl_433 manual.

**Comment by zuckschwerdt** [comment, marked as answer]
There is a note "Printed on the part as WH65B" and that sensor is fully supported in rtl_433.
Ambient Weather, Ecowitt, Froggit are known to resell Fine Offset.

**edit:** the packet is outlined here https://github.com/merbanan/rtl_433/blob/master/src/devices/fineoffset.c#L152

**Reply by Sandman2001** [reply]
Thank you sir!

**Reply by zuckschwerdt** [reply]
If all works well let us known the part number, we can add that to the description. PR for a line like https://github.com/merbanan/rtl_433/blob/master/src/devices/fineoffset.c#L154 welcome.

**Reply by Sandman2001** [reply]
Will have to confirm when I return home, but data sheet stated the sensor array operates at 915MHz, don't know how much that will change things and how easily configurable the software is to accommodate that.

**Reply by zuckschwerdt** [reply]
If your receiver supports a frequency all is well, rtl_433 is agnostic to the choosen frequency.

**Reply by Sandman2001** [reply]
Following up on this issue, I am still having issues with gathering data transmissions. I'll attach pictures below, but I am running both a WH32B and a WH65B. The transmissions for the 32B is intermittent, and the 65B is not getting picked up at all. I can however see the transmissions occur on the frequency/amplitude graph. I am currently running testing on SDRSharp. you can see the settings being ran in the screenshot
![setup_ws](https://github.com/user-attachments/assets/235e4a8f-41cf-4ae2-9f16-872ceaad705b)
![setup_computer](https://github.com/user-attachments/assets/fbba9623-c7af-4419-acbb-0fdd82744abb)
![SDRSharp_WS32B_grab](https://github.com/user-attachments/assets/907f2543-d1c0-4287-99ab-240d27d9b863)

**Reply by Sandman2001** [reply]
I will add that there are occasional Bit buffer warnings and errors, but have only been observed a few times.

**Reply by Sandman2001** [reply]
Update: I ran the windows CLI version and was able to read all of the devices
![image](https://github.com/user-attachments/assets/eac48e55-2da8-40e5-b956-8927ac012789)

A question I have from this: I obtained the base station receiver and now it seems everything is communicating. Do the sensor arrays need to communicate with the base station at least once to be configured properly? If so, is there opensource code already configured to relay this signal/communication steps to the WH65B sensors? Reason for this: Looking to scale this to multiple sensors and log data to a sql server via embedded micro controllers, goal is to be able to throw these sensors into a field and start working out of the box with as minimal setup as possible (preferably throw in batteries, secure to pole and leave).

**Reply by zuckschwerdt** [reply]
These sensors are send-only, there is no need for a receiver to configure things like encryption keys or such.

Best guess to the bad results in your earlier tests might be frequency issues (you need to tune close but not centered on the signal as there is attenuation close to the relative 0 Hz center) or maybe auto-gain trouble with the strong signals close by.

You can monitor both by using `-Y autolevel -M level -M noise`
